go-controller/pkg/node/controllers/egressip/egressip.go

```diff
@@ -0,0 +1,928 @@
+					podsToDelete = append(podsToDelete, podNamespacedName)
+				}
+			}
+			for _, podToDelete := range podsToDelete {
+				delete(targetPods, podToDelete)
+			}
+			if update == nil || update.targetNamespacesWithPods[targetNamespace] == nil {
+				namespacesToDelete = append(namespacesToDelete, targetNamespace)
+			}
+		}
+		for _, nsToDelete := range namespacesToDelete {
+			delete(existing.targetNamespaces, nsToDelete)
+		}
+	}
+	// clean up pod independent configuration first
+	// if EIP IP has changed and therefore could be hosted by a different interface, remove old EIP
+	// Delete addresses and routes under the following conditions
+	// 1. existing contains a non nil IP and update is nil
+	// 2. existing contains an ip and update contains an ip and update contains an ip different to existing
+	if (update == nil && existing.ip != nil) ||
+		(update != nil && update.ip != nil && existing.ip != nil && !existing.ip.Equal(*update.ip)) {
+		if err := c.linkManager.DelAddress(*existing.ip); err != nil {
+			// TODO(mk): if we fail to delete address, handle it
+			return fmt.Errorf("failed to delete egress IP address %s on link: %w", existing.ip, err)
+		}
+	}
+	if (update == nil && existing.route != nil) ||
+		(update != nil && update.route != nil && existing.route != nil && !existing.route.Equal(*update.route)) {
+		// route manager takes care of retry
+		c.routeManager.Del(*existing.eIP.route)
+	}
+
+	// apply new changes
+	if update != nil && update.ip != nil && update.route != nil {
+		for updatedTargetNS, updatedTargetPod := range update.targetNamespacesWithPods {
+			existingNs, found := existing.targetNamespaces[updatedTargetNS]
+			if !found {
+				existingNs = map[ktypes.NamespacedName]*podIPConfigList{}
+				existing.targetNamespaces[updatedTargetNS] = existingNs
+			}
+			for updatedPodNamespacedName, updatedPod := range updatedTargetPod {
+				existingTargetPodConfig, found := existingNs[updatedPodNamespacedName]
+				if !found {
+					existingTargetPodConfig = newPodIPConfigList()
+					existingNs[updatedPodNamespacedName] = existingTargetPodConfig
+				}
+				// applyPodConfig will apply pod specific configuration - ip rules and iptables rules
+				err := c.applyPodConfig(existingTargetPodConfig, update)
+				if err != nil {
+					return fmt.Errorf("failed to apply pod %s/%s configuration for EgressIP %s IP %s: %v",
+						updatedPod.Namespace, updatedPod.Name, update.name, update.ip.String(), err)
+				}
+			}
+		}
+		// TODO(mk): only apply the follow when its new config or when it failed to apply
+		// Ok to repeat requests to route manager and link manager
+		if err := c.linkManager.AddAddress(*update.ip); err != nil {
+			return fmt.Errorf("failed to add address EgressIP %s IP %s to link manager: %v", update.name,
+				update.ip.String(), err)
+		}
+		existing.ip = update.ip
+		// route manager manages retry
+		c.routeManager.Add(routemanager.RoutesPerLink{Link: update.route.Link,
```

Comment:
Shouldn't we use `update.route` here for consistency at least.

Reply:
yep - fixed.